Marlin/Marlin_main.cpp

```diff
@@ -1644,7 +1639,7 @@ static void clean_up_after_endstop_or_probe_move() {
 
     #if ENABLED(DELTA)
 
-      feedrate = XY_PROBE_FEEDRATE;
+      feedrate = (feed_rate != 0.0) ? feed_rate : XY_PROBE_FEEDRATE;
```

Comment:
Sounds good to me! Do you want to finish up this PR and submit it, or does it need more work yet?